addons/survey/models/survey_survey.py

```diff
@@ -397,6 +408,22 @@ def _onchange_session_speed_rating(self):
             survey.question_ids._update_time_limit_from_survey(
                 is_time_limited=survey.session_speed_rating, time_limit=survey.session_speed_rating_time_limit)
 
+    @api.onchange('restrict_user_ids', 'user_id')
+    def _onchange_restrict_user_ids(self):
+        """
+         Add survey user_id to restrict_user_ids when:
+         - restrict_user_ids is not False
+         - user_id is not part of restrict_user_ids
+         - user_id is not a survey manager
+        """
+        surveys_to_check = self.filtered(lambda s: s.restrict_user_ids and bool(s.user_id - s.restrict_user_ids))
+        users_are_managers = {
+            user: user.has_group('survey.group_survey_manager')
+            for user in surveys_to_check.user_id
+        }
```

Comment:
Would use a classic recordset, not sure why we have to use a dict ?
```suggestion
        manager_users = tocheck.user_id.filtered(lambda user: user.has_group('survey.group_survey_manager'))
```

Reply:
ok